### DISCUSSION TITLE
Desktop favorites not saved

### DISCUSSION THREAD

**Discussion by DSpammy** [devemux86/cruiser#268] [Q&A] [answered]
Hello Everybody!

I'm new to cruiser. I use it on a (manjaro) linux desktop. I created a favorite point today but it lost after exiting from cruiser. When I see .userPrefs/cruiser folder, there is a "favorites" folder in it and in the favorites there is a "prefs.xml" file. But in that file there are no information about favorites at all. It contains three lines only, same as the first three line in prefs.xml in cruiser (not favorites) folder. After exiting from cruiser "favorites" folder and the file in it disappears. Fortunately I can export favorites and import them next time but that is a suboptimal solution.

My java version is openjdk version "1.8.0_382".

Did I find a bug in cruiser or should I use another java runtime?

Thank you for your help in advance.

Attila

**Comment by devemux86** [comment]
Hi @DSpammy,

Do you use the latest desktop Cruiser?
https://github.com/devemux86/cruiser/releases

Arch Linux users can also install Cruiser from the AUR:
https://aur.archlinux.org/packages/cruiser

Favorites on desktop Cruiser now are stored in the `favorites.gpx` file inside app folder.
You can backup this file or move it to new Cruiser installations and favorites are loaded automatically.

I also use and work on Linux (Debian)!

**Reply by DSpammy** [reply]
Yes, it is the latest version, v 3.0.28 installed from AUR. I will try remove the package, delete cruiser folder and reinstall cruiser. Currently I don't see any app folder. Where should it exist?

**Reply by devemux86** [reply]
The AUR package is maintained by the community, so I do not know more details.

Can you try the zip from releases? You can extract it in your home folder.
https://github.com/devemux86/cruiser/releases

**Reply by DSpammy** [reply]
Done. The result is same, favorites are not reloaded after restart. Probably they were not saved.

**Reply by DSpammy** [reply]
Install from zip works! Thank you for your help.

**Comment by DSpammy** [comment]
I tested today the AUR package again. The problem is with favorites that I have no write permission on /usr/share/java/cruiser folder where the app wants to store favorites.gpx. I could change the permissions but what if an other user wants to create favorites? All users on the system will share same favorites? Or all of them should install cruiser locally? @devemux86, how do you use it on your debian system? I think that storing favorites together with the app is a wrong idea. Favorites should be stored somewhere within the user home directory instead.

**Reply by devemux86** [reply]
> how do you use it on your debian system?

I do not maintain a .deb package.

I use it with the zip in home folder (a long time ago on Windows and now on Linux).

> Favorites should be stored somewhere within the user home directory instead.

I can check it. :slightly_smiling_face:

**Comment by devemux86** [comment]
> Favorites should be stored somewhere within the user home directory instead.

It will be available in the next desktop Cruiser.
(along with the last route)

**Reply by DSpammy** [reply]
Great! Cruiser will be more better...

**Comment by devemux86** [comment, marked as answer]
> Favorites should be stored somewhere within the user home directory instead.

(also the route)

Implemented in desktop [Cruiser 3.0.29](https://github.com/devemux86/cruiser/releases).

**Reply by DSpammy** [reply]
Good news. I'm waiting for the AUR package. Hopefully it arrives shortly. Thanks.

Attila

**Reply by DSpammy** [reply]
Installed 3.0.29 from AUR and it really works. Thanks again.

Attila
